test/e2e/network/netpol/network_policy.go

```diff
@@ -0,0 +1,884 @@
+			reachability.ExpectPeer(&Peer{Namespace: nsX}, &Peer{}, false)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 81, Protocol: v1.ProtocolTCP, Reachability: reachability}, isVerbose)
+
+			ginkgo.By("Allowing all ports")
+
+			policyAllowAll := GetAllowEgress()
+			CreateOrUpdatePolicy(k8s, policyAllowAll, nsX, true)
+
+			reachabilityAll := NewReachability(model.AllPods(), true)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 81, Protocol: v1.ProtocolTCP, Reachability: reachabilityAll}, isVerbose)
+		})
+
+		ginkgo.It("should stop enforcing policies after they are deleted [Feature:Netpol]", func() {
+			ginkgo.By("Creating a network policy for the server which denies all traffic.")
+
+			// Deny all traffic into and out of "x".
+			policy := GetDenyAll("deny-all")
+			nsX, _, _, model, k8s := getK8SModel(f)
+			CreateOrUpdatePolicy(k8s, policy, nsX, true)
+			reachability := NewReachability(model.AllPods(), true)
+
+			// Expect all traffic into, and out of "x" to be False.
+			reachability.ExpectPeer(&Peer{Namespace: nsX}, &Peer{}, false)
+			reachability.ExpectPeer(&Peer{}, &Peer{Namespace: nsX}, false)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability}, isVerbose)
+
+			err := k8s.CleanNetworkPolicies(model.NamespaceNames)
+			time.Sleep(3 * time.Second) // TODO we can remove this eventually, its just a hack to keep CI stable.
+			framework.ExpectNoError(err, "unable to clean network policies")
+
+			// Now the policiy is deleted, we expect all connectivity to work again.
+			reachabilityAll := NewReachability(model.AllPods(), true)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachabilityAll}, isVerbose)
+		})
+
+		ginkgo.It("should allow egress access to server in CIDR block [Feature:Netpol]", func() {
+			// Getting podServer's status to get podServer's IP, to create the CIDR
+			nsX, nsY, _, model, k8s := getK8SModel(f)
+			podList, err := f.ClientSet.CoreV1().Pods(nsY).List(context.TODO(), metav1.ListOptions{LabelSelector: "pod=b"})
+			framework.ExpectNoError(err, "Failing to list pods in namespace y")
+			pod := podList.Items[0]
+
+			hostMask := 32
+			if utilnet.IsIPv6String(pod.Status.PodIP) {
+				hostMask = 128
+			}
+			podServerCIDR := fmt.Sprintf("%s/%d", pod.Status.PodIP, hostMask)
+			policyAllowCIDR := GetAllowEgressByCIDR("a", podServerCIDR)
+			CreateOrUpdatePolicy(k8s, policyAllowCIDR, nsX, true)
+
+			reachability := NewReachability(model.AllPods(), true)
+			reachability.ExpectAllEgress(NewPodString(nsX, "a"), false)
+			reachability.Expect(NewPodString(nsX, "a"), NewPodString(nsY, "b"), true)
+			ValidateOrFail(k8s, model, &TestCase{FromPort: 81, ToPort: 80, Protocol: v1.ProtocolTCP, Reachability: reachability}, isVerbose)
+		})
+
+		ginkgo.It("should enforce except clause while egress access to server in CIDR block [Feature:Netpol]", func() {
```

Comment:
this seems to work only for IPv4, /4 and /32 masks

Reply:
yeah, it will specifically just grab one of those IP addresses and verify it.

Reply:
yeah, but  those masks should be different for IPv6, I had to fix it in the other network policy tests
https://github.com/kubernetes/kubernetes/pull/93583

Reply:
ah this is a newer change, I thought we only tested ipv4, yes ill incorporate it now thx

Reply:
fixed, resolving